### PR TITLE
feat: pass data object to on-click handler

### PR DESCRIPTION
**Pull request by luqven**
This PR allows for the `ProductImageContainer.tsx` `onClick` handler to accept a `data` object. This object can then be passed onto a different component in the future.

This was done to try and share selection information between the images container and an upcoming IM modal in the following PR.

### CODE REVIEW

**Comment by luqven** [comment]
Current dependencies on/for this PR:
* next
  * **PR #184** 
    * **PR #185** 
      * **PR #187**  (this PR)
        * **PR #186** 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: frontend/src/components/ProductImage/ProductImageContainer.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Overlay } from "../layouts/Overlay";
 import styles from "./ProductImageContainer.module.scss";
 
 interface Props {
-  onOpenBreakoutClick: () => void;
+  onOpenBreakoutClick: (data: any) => void;
```
Comment (luqven) [inline frontend/src/components/ProductImage/ProductImageContainer.tsx:9]: TODO: `any` needs to be changed here to an actual interface/type delcarion.

File: frontend/src/components/ProductPageImages.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ export const ProductPageImages = ({
         <AddButton
           disabled={disabled}
           label="ADD IMAGE"
-          onOpenBreakoutClick={onClick}
+          onOpenBreakoutClick={() => onClick(null)}
```
Comment (luqven) [inline frontend/src/components/ProductPageImages.tsx:24]: When `Add image` is selected, no `data` should be passed onto the IM moda.

File: frontend/src/components/ProductImage/ProductImageContainer.tsx (hunk 1 above)
Comment (frederickfogerty) [reply frontend/src/components/ProductImage/ProductImageContainer.tsx:9]: 👍

**Review by frederickfogerty — COMMENTED** [review]
LGTM so far. I think it's definitely important to think about what data we want to pass around here. It's good to note here that we can store anything in the custom attribute (e.g. asset ids, sources, etc), so far it only depends on `src` being set. 

Some options I'm thinking of for what data to store:
- the minimal data set (e.g. asset id, source id)
- the whole asset data object
- trying to reverse set the asset id and source id from the `src` url. This sounds like a pain though.

Atm I'm leaning towards the minimal data set option. What do you think?

**Comment by luqven** [comment]
Closing this since UI and data flow have changed.

**Pull request closed without merging**